Yank/pipeline.py

```diff
@@ -1451,6 +1486,19 @@ def _setup_system(self, system_file_path, pack, alchemical_charge,
 
             # Solvate unit. Solvent models different than tip3p need parameter modifications.
             solvent_model = solvent['solvent_model']
+            # Check that solvent model has loaded the appropriate leap parameters.
+            # This does guarantee a failure, but is a good sign of it.
+            solvent_warning = None
+            if _OPENMM_LEAP_SOLVENT_FILES_MAP[solvent_model] not in loaded_water_files:
+                solvent_warning = ("WARNING: The solvent_model {} may not work for loaded "
+                                   "leaprc.water.X files.\n We expected {} to make your "
+                                   "solvent model work, but did not find it.\n "
+                                   "This is okay for tip4pew leaprc file with tip3p solvent_model, "
+                                   "but not the other way around.\nThis does "
+                                   "not mean the system will not build, but it may throw "
```

Comment:
Hmm. I'm worried that we risk clogging the output with warnings and errors so detailed since they get printed even if `verbose` is `False`.

This would totally be a separate issue/PR, but to keep our messages both compact and exhaustive, what do you think about having short error/warning messages that convey the point associated to numerical codes and have a page on the docs with more details on the errors in case the user is not already familiar with them?

Reply:
I like this idea, but I think this should be handled in a future PR, and when we go through and make better error handling more robust. Like I said, this is a bit of a crude fix, but it sets up the framework. I'll make an enhancement issue or this.